platforms/ios/src/TangramMap/TGMarker.mm

```diff
@@ -51,99 +62,148 @@ - (instancetype)initWithMapView:(TGMapViewController *)mapView
         type = TGMarkerType::none;
         self.visible = YES;
         self.drawOrder = 0;
-        self.map = mapView;
+        [self map:mapView];
     }
 
     return self;
 }
 
-- (void)setStylingString:(NSString *)styling
+- (NSError *)createNSError
+{
+    NSMutableDictionary* userInfo = [[NSMutableDictionary alloc] init];
+    // TODO: add enum for object error
+    // [userInfo setObject: forKey:@"TGMarker"];
+
+    return [NSError errorWithDomain:@"TGMarker"
+                               code:0
+                           userInfo:userInfo];
+}
+
+- (NSError *)stylingString:(NSString *)styling
 {
     _stylingString = styling;
     _stylingPath = nil;
 
-    if (!tangramInstance || !identifier) { return; }
+    if (!tangramInstance || !identifier) { return nil; }
```

Comment:
Yeah maybe an exception can be thrown for the user to the API to use it aptly, which is make sure a map instance is set before setting styling on a marker.